src/pathfinder.cpp

```diff
@@ -287,22 +278,32 @@ class Pathfinder {
 	bool          updateAllCosts(v3s16 ipos, v3s16 srcdir, int current_cost, int level);
 
 	/**
-	 * recursive try to find a patrh to destionation
-	 * @param ipos position to check next
-	 * @param srcdir positionc checked last time
-	 * @param total_cost cost of moving to ipos
-	 * @param level current recursion depth
+	 * try to find a path to destination using a heuristic function
+	 * to estimate distance to target (A* search algorithm)
+	 * @param isource start position (index pos)
+	 * @param idestination end position (index pos)
 	 * @return true/false path to destination has been found
 	 */
-	bool          updateCostHeuristic(v3s16 ipos, v3s16 srcdir, int current_cost, int level);
+	bool          updateCostHeuristic(v3s16 isource, v3s16 idestination);
 
 	/**
-	 * recursive build a vector containing all nodes from source to destination
+	 * build a vector containing all nodes from destination to source;
+	 * to be called after the node costs have been processed
 	 * @param path vector to add nodes to
-	 * @param pos pos to check next
-	 * @param level recursion depth
+	 * @param ipos initial pos to check (index pos)
+	 * @return true/false path has been fully built
 	 */
-	void          buildPath(std::vector<v3s16> &path, v3s16 pos, int level);
+	bool          buildPath(std::vector<v3s16> &path, v3s16 ipos);
+
+        /**
+         * go downwards from a position until some barrier
+         * is hit.
+         * @param pos position from which to go downwards
+         * @param max_down maximum distance to go downwards
+         * @return new position after movement; if too far down,
+                   pos is returned
+         */
+        v3s16         walkDownwards(v3s16 pos, unsigned int max_down);
```

Comment:
Please use tabs to indent, or tab + space for the comment above.

Reply:
https://github.com/minetest/minetest/pull/9339#issuecomment-578248679

In fact I would suggest merging this without fixing the code style. The much-needed bugfixes are more important than the (still) wrong code style of the entire file.